### PR TITLE
enhancement: Integrate HydraDX

### PR DESCRIPTION
**Pull request by maxwellfoley3**
## Description
Integrate Substrate chain HydraDX

Note: Should not be merged until a discussion around some lines on `shared/substrate.ts` is resolved.

To be merged with https://github.com/hicommonwealth/chain-events/pull/47

## Motivation and Context
Partnership with HydraDX

## How has this been tested?
Manually by ensuring page properly loads

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [ ] yes, and they are tested: [enter the % coverage here]
- [ ] yes, and they are not tested: [enter the % coverage here]
- [ ] yes, but I did not run tests
- [x] no

### CODE REVIEW

File: client/scripts/views/components/sidebar/index.ts [hunk 1]
```diff
@@ -203,7 +204,8 @@ export const OnchainNavigationModule: m.Component<{}, {}> = {
           contentRight: [], // TODO
         }),
       // proposals (substrate, cosmos, moloch only)
-      !app.community && ((app.chain?.base === ChainBase.Substrate && app.chain.network !== ChainNetwork.Darwinia)
+      !app.community && ((app.chain?.base === ChainBase.Substrate && app.chain.network !== ChainNetwork.Darwinia
+                        && app.chain.network != ChainNetwork.HydraDX)
```
Comment (jnaviask) [inline client/scripts/views/components/sidebar/index.ts:208]: Proposals page includes council motions, not just democracy. If Hydra has these, we'll want to make sure this link works.

File: server/migrations/20210301230133-add-hydradx.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction(async (t) => {
+      await queryInterface.bulkInsert('Chains', [{
```
Comment (jnaviask) [inline server/migrations/20210301230133-add-hydradx.js:6]: Needs `base: 'substrate'`.

File: client/scripts/controllers/chain/hydradx/main.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { ChainClass, NodeInfo } from 'models';
+import { IApp } from 'state';
+import Substrate from '../substrate/main';
+import SpecTypes from './spec';
+
+class HydraDX extends Substrate {
+  constructor(n: NodeInfo, app: IApp) {
+    super(n, app, ChainClass.HydraDX);
+    this.signaling.disable();
```
Comment (jnaviask) [inline client/scripts/controllers/chain/hydradx/main.ts:9]: I would check the hydra runtime to ensure that these are the only modules we need to disable.

**Review by jnaviask — CHANGES_REQUESTED** [review]
Couple of small things / questions, but basically looks good!

**Comment by dillchen** [comment]
@maxwellfoley3 for future reference in the `Address` Model, one needs to add the chain ID to address verification. 

The chain ID is mentioned here. https://github.com/galacticcouncil/HydraDX-node/blob/master/node/src/chain_spec.rs

Max, it'd also be good to make sure you have Polkadot Extension downloaded so you can go through the full address linking flow, which is a necessary part of making sure chain integrations work in the future :)

**Comment by dillchen** [comment]
@maxwellfoley3 if @jnaviask's comments have been resolved, we can merge this

**Comment by dillchen** [comment]
(after merge conflicts have been taken care of ofc)

File: client/scripts/controllers/chain/hydradx/main.ts (hunk 3 above)
Comment (dillchen) [reply client/scripts/controllers/chain/hydradx/main.ts:9]: seems resolved

**Review by dillchen — APPROVED** [review]

**Review by dillchen — APPROVED** [review]

**Pull request merged**